src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -216,7 +216,11 @@ private string GenerateCrossgen2ResponseFile()
                 result.AppendLine($"--out:\"{_outputR2RImage}\"");
                 if (!String.IsNullOrEmpty(Crossgen2ExtraCommandLineArgs))
                 {
-                    result.AppendLine(Crossgen2ExtraCommandLineArgs);
+                    foreach (string extraArg in Crossgen2ExtraCommandLineArgs.Split(';'))
+                    {
+                        if (!String.IsNullOrEmpty(extraArg))
```

Comment:
If you pass `StringSplitOptions.RemoveEmptyEntries`, you may skip checking for the empty string.